### PR TITLE
ME Pattern Buffer (Gregic Energistics ME IO Buffer CEU Port)

### PR DESCRIPTION
**Pull request by Ghostipedia**
Before Anyone asks, yes permission to use GE code was granted, I feel we should still give credit in the readme to Qwer and the EpimorphicPioneers Team for some of the great work they've done so far, even if permission was granted.
image

Currently still working through it, this PR will server as yet another "Ghost needs a checklist or they will forget" thing, as well as let people view my (arguably awful) porting skills :tr: 

TODO : 
- [x] Get the Game to load
- [x] Get the Actual block to exist
- [x] Functionality from GE ported
~~- [ ] Removal of Uneeded behaviors and or Feature Rebalancing (Auto-output back from itself, others TBD)~~
^This is very susso and out of my skillset to detangle. Will need addressing at a future date.
- [x] Proper Textures for UI, Overlay, and Hull/casing Render
- [x] Lang, so much lang
- [x] IF YOU FORGET JADE SUPPORT I WILL CRY
~~- [ ] Put a warning about cross pattern contamination as how regular machines behave, and that it will be fixed eventually~~
^ In regards to this just assume they conflict like how LCRs and other machines are. We **have** do do the distinction API come 1.4.1+ urgently to improve the functionality of Dual Hatches and the ME Pattern Buffer
- [x] Actual Recipe (LuV Assline)
- [x] Any other Things Reviewers of this PR might fight horrendous and leave as review (Calling it now I will have to reoganize the location of some classes most likely.)

DONE :) HUGE Thank you to Onion and Screret with the help with the proxy hatch, that thing was evil!

GOAL : 
Create a Multipart for GTCEU multiblocks that allows direct insertion of ME Patterns in its internal inventory. Allowing for easier automation when in combination with ME output Bus/Hatches. This Multipart will be usable from Post Assline progression, however if other devs feel it is still hilariously too strong, we can move it to post Mk1 Fusion and require Ruthenium.
This PR also servers to help aid #1731 from @screret migrate GE parts of old to present without any issues.

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/utils/GTNBTUtils.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/utils/GTNBTUtils.java]: These already exist in GTUtil

**Comment by Ghostipedia** [comment]
I strongly have a love/hate relationship with spotless 🙃

**Comment by Ghostipedia** [comment]
Today we learn a valuable lesson.
Lombok is evil. static final vars and lombok and super inheretance I hate I hate I hate I hate I hate I hate

**Comment by Ghostipedia** [comment]
This post was sponsored by 3-4hours of painful DEBUGGING A VERY """OBVIOUS""" ISSUE

**Comment by Ghostipedia** [comment]
A Visual Example of the new textures

https://github.com/user-attachments/assets/178a69ca-c6f1-411c-84d6-67b932b59882

File: src/main/java/com/gregtechceu/gtceu/api/machine/fancyconfigurator/MEFancyInvConfigurator.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/api/machine/fancyconfigurator/MEFancyInvConfigurator.java]: just name this `FancyInvConfigurator`, since it isn't specific to ME parts in any way.

File: src/main/java/com/gregtechceu/gtceu/api/machine/fancyconfigurator/MEFancyTankConfigurator.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/api/machine/fancyconfigurator/MEFancyTankConfigurator.java]: name this `FancyTankConfigurator`

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferPartMachine.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferPartMachine.java]: just remove this?

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferPartMachine.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferPartMachine.java]: this TODO should probably be done?

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferPartMachine.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferPartMachine.java]: remove this newline

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferProxy.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferProxy.java]: remove the initialization, it goes unused anyway and can break the saving logic.

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/trait/MEPatternBufferRecipeHandler.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/trait/MEPatternBufferRecipeHandler.java]: remove, it's useless.

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/trait/MEPatternBufferRecipeHandler.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/trait/MEPatternBufferRecipeHandler.java]: this should return `0.0D`

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/trait/MEPatternBufferRecipeHandler.java
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/trait/MEPatternBufferRecipeHandler.java]: this should return `0.0D`

**Review by screret — CHANGES_REQUESTED** [review]

**Comment by Ghostipedia** [comment]
Wahoo! Checks Passed, this is good to go!

**Review by screret — APPROVED** [review]
wait for approval from someone else as well if possible.
also, testing.

**Pull request merged**
